include/bio/format/vcf_input_handler.hpp

```diff
@@ -0,0 +1,730 @@
+     */
+    static size_t parse_dynamic_type(var_io::dynamic_type_id const  id,
+                                     std::string_view const         input_string,
+                                     detail::is_dynamic_type auto & output); // implementation after class
+
+    //!\brief Parse the CHROM field. Reading chrom as number means getting the index (not converting string to number).
+    template <typename parsed_field_t>
+    void parse_field(vtag_t<field::chrom> const & /**/, parsed_field_t & parsed_field)
+    {
+        std::string_view raw_field = get<field::chrom>(raw_record);
+
+        if (raw_field != last_chrom)
+        {
+            if (auto it = header.string_to_contig_pos().find(raw_field);
+                it == header.string_to_contig_pos().end()) // contig name was not in header, insert!
+            {
+                header.contigs.push_back({.id = static_cast<std::string>(raw_field)});
+                header.add_missing();
+
+                last_chrom_idx = header.contigs[header.contigs.size() - 1].idx;
+
+                if (print_warnings)
+                {
+                    seqan3::debug_stream << "[bio::var_io::warning] The contig name \"" << raw_field
+                                         << "\" found on line " << line << " is not present in the header.\n";
+                }
+            }
+            else
+            {
+                last_chrom_idx = header.contigs[it->second].idx;
+            }
+
+            last_chrom = raw_field;
+        }
+
+        if constexpr (std::integral<parsed_field_t>)
+            parsed_field = static_cast<parsed_field_t>(last_chrom_idx);
+        else
+            parsed_field = static_cast<parsed_field_t>(raw_field);
+    }
+
+    /* POS, ID, REF are handled correctly by default */
+
+    //!\brief Overload for parsing ALT.
+    template <typename parsed_field_t>
+        requires detail::back_insertable<parsed_field_t> &&
+          std::ranges::range<std::ranges::range_reference_t<parsed_field_t>>
+    void parse_field(vtag_t<field::alt> const & /**/, parsed_field_t & parsed_field)
+    {
+        std::string_view raw_field = get<field::alt>(raw_record);
+
+        if (raw_field != ".")
+        {
+            for (std::string_view subfield : raw_field | detail::eager_split(','))
+            {
+                parsed_field.emplace_back();
+                // delegate parsing of element to base
+                parse_field_aux(subfield, parsed_field.back());
+            }
+        }
+    }
+
+    //!\brief Overload for parsing QUAL.
+    template <seqan3::arithmetic parsed_field_t>
+    void parse_field(vtag_t<field::qual> const & /**/, parsed_field_t & parsed_field)
+    {
+        std::string_view raw_field = get<field::qual>(raw_record);
+
+        if (raw_field == ".")
+        {
+            parsed_field = var_io::missing_value<parsed_field_t>;
+        }
+        else
+        {
+            parsed_field = parsed_field_t{};
```

Comment:
why do you need to default initialise beforehand?

Reply:
Not sure 😅